pkg/providers/provider_test.go

```diff
@@ -34,15 +34,6 @@ func TestNewProvider(t *testing.T) {
 			},
 			wantErr: true,
 		},
-		{
```

Comment:
@msvticket Instead of just removing the test case, we should add test cases for both scenarios i.e. to be successful with secrets and without secrets.

Reply:
I've put the test back. Since there already exists test cases for empty or missing secrets I just made them pass.